File: Programmer_test/PT_Java/P77884.java
public class P77884 {
    public static void main(String[] args) {
        P77884 T = new P77884();
        System.out.println(T.solution(13, 17));
        System.out.println(T.solution(24, 27));
    }

    public int solution(int left, int right) {
        int answer = 0;
        int[] divisor = new int[right+1];
        for (int i = 0; i <= right; i++) {
            divisor[i] = 1;
        }
        int temp = 0;
        for (int i = 2; i <= right; i++) {
            temp = i;
            while (temp <= right) {
                divisor[temp] += 1;
                temp += i;
            }
        }
        for (int i = left; i <= right; i++) {
            if ((divisor[i] % 2) == 0) {
                answer += i;
            }
            else {
                answer -= i;
            }
        }
        return answer;
    }
}

File: Programmer_test/PT_Python/P77884.py
def solution(left, right):
    answer = 0
    # 1. right 크기만큼의 약수 개수 저장 배열 선언
    divisor = [0] * (right + 1)
    
    # 2. 모든 숫자는 기본적으로 1을 약수로 가지므로 1로 초기화
    for i in range(0, right + 1):
        divisor[i] = 1
        
    temp = 0
    # 3. 에라토스테네스의 체와 유사하게 배수들의 약수 개수를 누적 카운팅
    for i in range(2, right + 1):
        temp = i
        while temp <= right:
            divisor[temp] += 1
            temp += i  # i의 배수로 이동
            
    # 4. left부터 right까지 돌며 짝수면 더하고, 홀수면 빼기
    for i in range(left, right + 1):
        if (divisor[i] % 2) == 0:
            answer += i
        else:
            answer -= i
            
    return answer

print(solution(13, 17))
print(solution(24, 27))